.github/workflows/build_and_upload.yml

```
@@ -11,36 +11,36 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-20.04, windows-2019, macos-11]
+        os: [ubuntu-22.04, windows-2022, macos-13]
```

You may want to consider
```suggestion
        os: [ubuntu-22.04, windows-2022, macos-13, macos-14]
```

To support Apple Silicon native binaries too